portfolio/src/main/webapp/game.js

```diff
@@ -100,10 +117,106 @@ class Board {
       } else {
         this.board[row][col] = 2;
       }
+      this.numTiles++;
+      this.checkGameState(this.redTrue);
       this.redTrue = !this.redTrue;
+    } else {
+        alert("Cannot place tile there. Try again.");
+    }
+  }
+  
+  /*
+  * Checks whether move made by redTrue player resulted in win or if game board is full.
+  * Sends alert with win message or gameOver message if board is full and allows user to
+  * restart game.  
+  */
+  checkGameState(redTrue) {
+    var player = 0;
+    if (redTrue) {
+      player = 1; 
+    } else {
+        player = 2;
+    }
+    var win = false; 
+
+    // check rows
+    for (var i = 0; i < this.board.length; i++) {
+      var rowWin = true;
+      for (var j = 0; j < this.board[i].length; j++) {
+        if (this.board[i][j] != player) {
+          rowWin = false;
+        }
+      }
+      if (rowWin) {
+        win = true;
+        break;
+      }
+    }
+
+    // check cols 
+    for (var j = 0; j < this.board[0].length; j++) {
+      var colWin = true;
+      for (var i = 0; i < this.board.length; i++) {
+        if (this.board[i][j] != player) {
+          colWin = false;
+        }   
+      }
+      if (colWin) {
+        win = true;
+        break;
+      }
+    }
+
+    // check diaganol
+    var winDiagUp = true;
+    for (var i = 0; i < this.board.length; i++) {
+        if (this.board[i].length != this.board.length) {
+            console.log("Error board not square");
+            break;
+        }
+        if (this.board[i][i] != player) {
+            winDiagUp = false; 
+        }
+    }
+    var winDiagDown = true;
+    var j = 0;
+    for (var i = this.board.length - 1; i > -1; i--) {
+        if (this.board[i].length != this.board.length) {
+            console.log("Error board not square");
+            break;
+        }
+        if (this.board[i][j] != player) {
+            winDiagDown = false; 
+        }
+        j++;
+    }
+
+    // show win message
+    if (win || winDiagUp || winDiagDown) {
```

Comment:
Nit:
Consider returning from the function and returning as soon as you know a win has happened. Currently, you continue to check cols and diagonal even if rows shows a win. Not a big deal here since the board is small, but is good practice for when you are dealing with bigger data sizes.